fieldservice/models/fsm_order.py

```diff
@@ -306,37 +314,41 @@ def onchange_scheduled_duration(self):
             self.scheduled_date_end = str(date_to_with_delta)
 
     def copy_notes(self):
+        old_desc = self.description
         self.description = ""
+        self.location_directions = ""
         if self.type and self.type.name not in ['repair', 'maintenance']:
             for equipment_id in self.equipment_ids:
                 if equipment_id:
-                    if equipment_id.notes is not False:
-                        if self.description is not False:
+                    if equipment_id.notes:
+                        if self.description:
                             self.description = (self.description +
                                                 equipment_id.notes + '\n ')
                         else:
                             self.description = (equipment_id.notes + '\n ')
         else:
             if self.equipment_id:
-                if self.equipment_id.notes is not False:
-                    if self.description is not False:
+                if self.equipment_id.notes:
+                    if self.description:
                         self.description = (self.description +
                                             self.equipment_id.notes + '\n ')
                     else:
                         self.description = (self.equipment_id.notes + '\n ')
         if self.location_id:
             s = self.location_id.direction
-            if s is not False and s != '<p><br></p>':
+            if s and s != '<p><br></p>':
                 s = s.replace('<p>', '')
                 s = s.replace('<br>', '')
                 s = s.replace('</p>', '\n')
-                if self.location_directions is not False:
-                    self.location_directions = (self.location_directions +
-                                                '\n' + s + '\n')
-                else:
-                    self.location_directions = (s + '\n ')
+                self.location_directions = (s + '\n ')
         if self.template_id:
-            self.todo = self.template_id.instructions
+            if self.template_id.instructions:
+                self.todo += self.template_id.instructions
```

Comment:
This is causing a duplicate of the same instructions or appends a new set of instructions leaving the previous template instructions, which seems incorrect to me.  With a new template comes a new set of instructions

And in the case of nothing in self.todo it raises error when trying to set a template on the form

 ` File "/home/odoo/build/OCA/field-service/fieldservice/models/fsm_order.py", line 346, in copy_notes
    self.todo += self.template_id.instructions
TypeError: unsupported operand type(s) for +=: 'bool' and 'str'`